internal/services/azurestackhci/stack_hci_virtual_hard_disk_resource_test.go

```diff
@@ -0,0 +1,231 @@
+package azurestackhci_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/azurestackhci/2024-01-01/virtualharddisks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type StackHCIVirtualHardDiskResource struct{}
+
+func TestAccStackHCIVirtualHardDisk_basic(t *testing.T) {
+	if os.Getenv(customLocationIdEnv) == "" {
+		t.Skipf("skipping since %q has not been specified", customLocationIdEnv)
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_stack_hci_virtual_hard_disk", "test")
+	r := StackHCIVirtualHardDiskResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccStackHCIVirtualHardDisk_complete(t *testing.T) {
+	if os.Getenv(customLocationIdEnv) == "" {
+		t.Skipf("skipping since %q has not been specified", customLocationIdEnv)
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_stack_hci_virtual_hard_disk", "test")
+	r := StackHCIVirtualHardDiskResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccStackHCIVirtualHardDisk_update(t *testing.T) {
```

Comment:
it only tests add/remove `tags`. Could you also test update `tags`?

Reply:
updated